src/core/inductive.ml

```diff
@@ -125,44 +190,51 @@ let fold_cons_typ (pos : popt) (codom : term list -> 'b list -> 'a -> 'c)
     ind_T: Π(p:Π(i1:A1),...,Π(im:Am), T i1 ... im -> Prop), U1 -> ... -> Un ->
     (Π(i1:A1),...,Π(im:Am), Π(t:T i1 ... im), π(p i1 ... im t))
     where Ui is the clause associated to the constructor ci. *)
-let gen_rec_type : Sig_state.t -> popt -> sym -> sym list -> term =
-  fun ss pos ind_sym cons_list ->
+let gen_rec_type :
+      Sig_state.t -> popt -> (sym * sym list) list
+      -> (term list * (sym * (tvar * tbox)) list) =
```

Comment:
Please explain what is returned. Would it be possible to structure this function as follows:
```
let gen_rec_types ss popt ind_list =
  let sym_pred_map = ... in
  let gen_rec_type : sym * sym list -> term = ... in
  (List.map gen_rec_type ind_list, sym_pred_map)
```

Reply:
> Please explain what is returned.

I don't understand: it is done.

Reply:
The function returns 2 things. You explain the first one but not the second.